src/stack.js

```diff
@@ -13,12 +13,32 @@ export default class Stack {
     this.size = 0
   }
 
-  push(value) {
-    this.top = new Node(value, this.top)
-    this.size++
+  isEmpty() {
+    return this.size === 0
   }
 
   length() {
     return this.size
   }
+
+  peek() {
+    return this.size === 0
+      ? null
+      : this.top.data
+  }
+
+  pop() {
+    const current = this.top
+    return this.size === 0
+      ? null
+      :(this.top = current.next,
```

Comment:
the way this is written is really hard for me to read and understand. I would have written it like this:

```js
pop() {
  if (this.size === 0) return null
  const current = this.top
  this.top = current.next
  this.size--
  return current.data
}
```